File: spec/java/src/io/kaitai/struct/spec/TestOpaqueWithParam.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.OpaqueWithParam;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestOpaqueWithParam extends CommonSpec {
    @Test
    public void testOpaqueWithParam() throws Exception {
        OpaqueWithParam r = OpaqueWithParam.fromFile(SRC_DIR + "term_strz.bin");

        assertEquals(r.one().buf(), "foo|b");
        assertEquals(r.one().trailer().intValue(), 0x61);
    }
}

File: spec/python/spec/test_opaque_with_param.py
import unittest
from testformats.opaque_with_param import OpaqueWithParam

class TestOpaqueWithParam(unittest.TestCase):
    def test_opaque_with_param(self):
        with OpaqueWithParam.from_file('src/term_strz.bin') as r:
            self.assertEqual(r.one.buf, "foo|b")
            self.assertEqual(r.one.trailer, 0x61)
